PR: https://github.com/odoo/odoo/pull/87082

From: 78d964b29d816a931c35dada7fc1e9c4327cc4a0
From: Kevin Baptiste
Date: 2022-04-04 15:48:29

Structural Changes: 4.9
Total Changes: 7

[IMP] hr: simplify signature requests on Plan

Having both an activity type "Request Signature" and "Signature
Request" on the plan were confusing and misleading for the users.

Related odoo/enterprise#25511
Related odoo/upgrade#3398

closes odoo/odoo#87082

Taskid: 2796407
Signed-off-by: Kevin Baptiste <kba@odoo.com>

================================= pseudo patch: =================================

--- a/addons/hr/models/hr_plan.py
+++ b/addons/hr/models/hr_plan.py
@@ -11,8 +11,8 @@ class HrPlan(models.Model):
     name = fields.Char('Name', required=True)
     company_id = fields.Many2one(
         'res.company', default=lambda self: self.env.company)
-    plan_activity_type_ids = fields.Many2many(
-        'hr.plan.activity.type',
+    plan_activity_type_ids = fields.One2many(
+        'hr.plan.activity.type', 'plan_id',
         string='Activities',
         domain="[('company_id', '=', company_id)]")
     active = fields.Boolean(default=True)

--- a/addons/hr/models/hr_plan_activity_type.py
+++ b/addons/hr/models/hr_plan_activity_type.py
@@ -26,9 +26,10 @@ class HrPlanActivityType(models.Model):
         ('other', 'Other')], default='employee', string='Responsible', required=True)
     responsible_id = fields.Many2one(
         'res.users',
-        'Name',
+        'Other Responsible',
         check_company=True,
         help='Specific responsible of activity if not linked to the employee.')
+    plan_id = fields.Many2one('hr.plan')
     note = fields.Html('Note')
 
     @api.depends('activity_type_id')
